tamr_client/attribute/_attribute.py

```diff
@@ -50,21 +59,23 @@ class ReservedName(TamrClientException):
     pass
 
 
-def by_resource_id(session: Session, dataset: Dataset, id: str) -> Attribute:
+def by_resource_id(
+    session: Session, parent: Union[Dataset, Project], id: str
```

Comment:
You could define a type alias at the top of this file like `Parent = Union[Dataset, Project]` and then use `Parent` in the function signatures, just to consolidate the definition of the parent type.